proposals/Level-1.md

```diff
@@ -203,80 +176,93 @@ corresponding exception.
-`rethrow` instruction.
+Note that a caught exception can be rethrown using the `rethrow` instruction.
 
 ### Rethrowing an exception
 
 The `rethrow` instruction takes the exception associated with the `except_ref`
-on top of the stack, and rethrows that exception. A rethrow has the same effect
+on top of the stack, and rethrows the exception. A rethrow has the same effect
 as a throw, other than an exception is not created. Rather, the referenced
 exception on top of the stack is popped and then thrown.
 
 ### Exception data extraction
 
-The `if_except block` defines a conditional query of the exception in the
-except_ref on top of the stack. The except_ref is popped when queried. The
-if_except block has two subblocks, the `then` and `else` subblocks, like that of
-an `if` block. The then block is a sequence of instructions following the
-`if_except` instruction. The else block begins with the `else` instruction.  The
-scope of the if_except block is from the `if_except` instruction to the
-corresponding `end` instruction.
+The `if_except` block begins with an `if_except` instruction, and
+has two instruction blocks, defined by the `then` and `else` instructions
+like that of an `if` block.
+
+The `if_except` instruction queries of the exception on top of the stack, and
+then pops the exception. If the exception has the same exception tag as defined
+by the `if_except` instruction, the `then` block is entered. Otherwise the
+`else` block is entered. If the if_except block doesn't have an else block,
+control passes to the instruction following that block.
+The exception value is not popped when queried.
 
 That is, the forms of an if_except block is:
 
 ```
+if_except block_type except_index
+  Instruction*
+end
+
 if_except block_type except_index
   Instruction*
 else
   Instruction*
 end
 ```
 
-The conditional query of an exception succeeds when the exception in the
-except_ref on the top of the stack has the corresponding exception tag (defined
-by `except_index`). After the query, the except_ref is popped from the stack.
+The conditional query of an exception succeeds when the exception on the top of
+the stack has the corresponding exception tag (defined by `except_index`).
 
 If the query succeeds, the values (associated with the exception value) are
 extracted and pushed onto the stack, and control transfers to the instructions
-in the then block. If the query fails, control is transfertd to the else block.
+in the then block.
+
+If the query fails, it either enters the else block, or transfer control to the
+end of the if_except block if there is no else block.
 
 ### Stack traces
 
-When an exception is thrown, conceptually a runtime stack is associated with the
-exception. However, WebAssembly does not have (direct) access to the stack
-trace. Hence, it is left to the embedder to maintain and use at it sees fit.
+Earlier discussion implied that when an exception is thrown, the runtime will
```

Comment:
You can probably drop the reference to earlier discussion. I feel like this document should capture the current state of the proposal rather than the history.